scripts/run_sft.py

```diff
@@ -117,7 +117,7 @@ def main():
         torch_dtype=torch_dtype,
         use_cache=False if training_args.gradient_checkpointing else True,
         device_map=get_kbit_device_map() if quantization_config is not None else None,
-        quantization_config=quantization_config,
+        quantization_config=quantization_config.to_dict(),
```

Comment:
Not needed, right?

```suggestion
        quantization_config=quantization_config,
```

Reply:
Anyway I do agree with @kashif that the best may be to just return a dictionary instead 👍🏻